backend/src/apiserver/server/util.go

```diff
@@ -268,3 +274,57 @@ func CheckPipelineVersionReference(resourceManager *resource.ResourceManager, re
 
 	return &pipelineVersionId, nil
 }
+
+func GetUserIdentity(ctx context.Context) (string, error) {
+	if ctx == nil {
+		return "", nil
+	}
+	md, _ := metadata.FromIncomingContext(ctx)
+	// If the request header contains the user identity, requests are authorized
+	// based on the namespace field in the request.
+	if userIdentityHeader, ok := md[common.GoogleIAPUserIdentityHeader]; ok {
+		if len(userIdentityHeader) != 1 {
+			return "", util.NewBadRequestError(errors.New("Request header error: unexpected number of user identity header. Expect 1 got "+strconv.Itoa(len(userIdentityHeader))),
+				"Request header error: unexpected number of user identity header. Expect 1 got "+strconv.Itoa(len(userIdentityHeader)))
+		}
+		userIdentityHeaderFields := strings.Split(userIdentityHeader[0], ":")
+		if len(userIdentityHeaderFields) != 2 {
+			return "", util.NewBadRequestError(errors.New("Request header error: user identity value is incorrectly formatted"),
+				"Request header error: user identity value is incorrectly formatted")
+		}
+		return userIdentityHeaderFields[1], nil
+	}
+	return "", nil
+}
+
+func GetNamespaceFromResourceReferences(resourceRefs []*api.ResourceReference) string {
+	namespace := ""
+	for _, resourceRef := range resourceRefs {
+		if resourceRef.Key.Type == api.ResourceType_NAMESPACE {
+			namespace = resourceRef.Key.Id
+			break
+		}
+	}
+	return namespace
+}
+
+func Authorize(resourceManager *resource.ResourceManager, userIdentity string, namespace string) (bool, error) {
+	// Authorization only happens when the userIdentity exists in the request header
+	// and it is the kubeflow deployment, which deploys the KFAM.
+	if len(userIdentity) != 0 && common.IsKubeflowDeployment() {
```

Comment:
when common.IsKubeflowDeployment() is called, we should make sure userIdentity is not 0 and namespace is not empty and return false if that's not the case. 
Otherwise user can bypass auth either by not specifying namespace or useridentity. Would be a security hole.

Reply:
Done. I enabled the authz when it is kubeflow deployment flag is enabled. We can also add another flag for multi-user in case people deploy the kubeflow but do not need multi-user authz. WDYT?

Reply:
If that's the case, can we change the flag to multi-user? I think we eventually want to support multi-user for non-kf deployment too.

Reply:
Changed to multi-user mode.